lib/utils.js

```diff
@@ -65,8 +66,22 @@ function uriAbs (req) {
   return req.protocol + '://' + req.get('host')
 }
 
+/**
+ * Composes and returns the fully-qualified URI for the request, to be used
+ * as a base URI for RDF parsing or serialization. For example, if a request
+ * is to `Host: example.com`, `GET /files/` using the `https:` protocol,
+ * then:
+ *
+ *   ```
+ *   uriBase(req)  // -> 'https://example.com/files/'
+ *   ```
+ *
+ * @param req {IncomingMessage}
+ *
+ * @return {string}
+ */
 function uriBase (req) {
-  return uriAbs(req) + (req.baseUrl || '')
+  return uriAbs(req) + url.resolve(req.baseUrl, req.path)
```

Comment:
Similarly, `uriAbs` doesn't seem intuitive; I actually think this one should be called `uriBase`.